src/pages/SignupPage.jsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from 'react';
+import styles from './SignupPage.module.css';
+import { Input } from '../components/common/input/Input';
+import { Button } from '../components/common/button/button';
+import { useDispatch, useSelector } from 'react-redux';
+import { signup as signupAction } from '../store/user/actions';
+import Logo from '../components/Navigation/Logo';
+
+function SignupPage() {
+    const [username, setUsername] = useState('');
```

Comment:
add react-hook-forms for the form

Reply:
Since this library wasn't mentioned anywhere in the original assignment, I'll need an extended deadline for this additional feature.